parquet-hadoop/src/test/java/org/apache/parquet/statistics/RandomValueGenerators.java

```diff
@@ -0,0 +1,395 @@
+      // Maximum may be a random number (which may be negative).
+      return random.nextInt(Math.abs(maximum) + 1);
+    }
+
+    public long randomLong() { return random.nextLong(); }
+    public long randomLong(long maximum) { return randomLong() % maximum; }
+
+    public float randomFloat() { return random.nextFloat(); }
+    public float randomFloat(float maximum) { return random.nextFloat() % maximum; }
+
+    public double randomDouble() { return random.nextDouble(); }
+    public double randomDouble(double maximum) { return random.nextDouble() % maximum; }
+
+    public BigInteger randomInt96(Random random) {
+      return new BigInteger((8 * 12), random);
+    }
+
+    public BigInteger randomInt96(BigInteger maximum, Random random) {
+      BigInteger result;
+      while ((result = randomInt96(random)).compareTo(maximum) > 0);
+      return result;
+    }
+
+    public char randomLetter() {
+      return ALPHABET.charAt(randomInt() % ALPHABET.length());
+    }
+
+    public String randomString(int maxLength) {
+      return randomFixedLengthString(randomInt(maxLength));
+    }
+
+    public String randomFixedLengthString(int length) {
+      StringBuilder builder = new StringBuilder();
+      for (int index = 0; index < length; index++) {
+        builder.append(randomLetter());
+      }
+
+      return builder.toString();
+    }
+
+    public int randomInt(int low, int high) {
+      int range = (high - low);
+      return low + randomInt(range);
+    }
+
+    protected abstract TValueType generateValue(TValueType minimum, TValueType maximum);
+
+
+    @Override
+    public Iterator<TValueType> iterator() {
+      return new ValueIterator<TValueType>(total) {
+        @Override
+        public TValueType next() {
+          return generateValue(minimum, maximum);
+        }
+      };
+    }
+  }
+
+  public static abstract class RandomBinaryIterable<T extends Comparable<T>> extends RandomValueGenerator<T> {
+    protected RandomBinaryIterable(long seed, PrimitiveType.PrimitiveTypeName forType, int total) {
+      super(seed, forType, total);
+    }
+
+    protected RandomBinaryIterable(long seed, PrimitiveType.PrimitiveTypeName forType, int total, int length) {
+      super(seed, forType, total, length);
+    }
+
+    protected RandomBinaryIterable(long seed, PrimitiveType.PrimitiveTypeName forType, int total, T minimum, T maximum) {
+      super(seed, forType, total, minimum, maximum);
+    }
+
+    public RandomBinaryIterable(long seed, PrimitiveType.PrimitiveTypeName forType, int total, T minimum, T maximum, int int96Length) {
+      super(seed, forType, total, minimum, maximum, int96Length);
+
+    }
+
+    public abstract byte[] generateValueAsByteArray(T minimum, T maximum);
+
+    byte[] reusedData = new byte[super.bufferLength];
+    private Binary asReusedBinary(byte[] data) {
+      Arrays.fill(reusedData, (byte)0);
+      System.arraycopy(data, 0, reusedData, 0, Math.min(data.length, reusedData.length));
```

Comment:
This should copy up to the length of data, not the reused array, right? Similarly, the reused array is only valid through the length of data, so you should use the factory method where you pass the length.